src/charm.py

```diff
@@ -240,13 +240,44 @@ def configure_app_non_leader(self, event: ops.framework.EventBase) -> None:
         if isinstance(event, MicroClusterNodeAddedEvent):
             self.cluster_nodes.join_node_to_cluster(event)
 
+    def _get_space_subnet(self, space: str):
+        """Get the first available subnet in the network space."""
+        space_nets = self.model.get_binding(binding_key=space).network.interfaces
+        if space_nets:
+            return space_nets[0].subnet
```

Comment:
I wonder about the behaviour if we have mult. subnets. Maybe we should log the fact we're picking the first one and ignoring the others if several are present?

Reply:
Juju spaces do not guarantee affiliation to any specific subnet. Also, if you do a `self.model.get_binding(binding_key=space).network.bind_address` it will return the first available address from the logical grouping of the juju space.

Reply:
We log the `public-network, cluster-network and the mon-ip` once they are all fetched, so i think logging them individually is no different ?